src/CoreWCF.Kafka/src/CoreWCF/Channels/KafkaBinding.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Confluent.Kafka;
+
+namespace CoreWCF.Channels
+{
+    public class KafkaBinding : Binding
+    {
+        private KafkaTransportSecurity _security = new();
+        private readonly KafkaTransportBindingElement _transportBindingElement = new();
+        private readonly BinaryMessageEncodingBindingElement _binaryMessageEncodingBindingElement = new();
+        private readonly TextMessageEncodingBindingElement _textMessageEncodingBindingElement = new();
+
+        public KafkaBinding(KafkaDeliverySemantics deliverySemantics = KafkaDeliverySemantics.AtLeastOnce)
+        {
```

Comment:
You mentioned about wanting to support TransportWithMessageCredentials. As that's the case, the general convention is for the binding constructor to take a SecurityMode value. You could have multiple constructors, one which takes just a SecurityMode, another which take the delivery semantics, and one which takes both. You can probably get that down to 2 overloads with using optional parameters.

Reply:
done